src/Authenticator/HttpDigestAuthenticator.php

```diff
@@ -78,33 +78,31 @@ public function __construct(IdentifierInterface $identifier, array $config = [])
     public function authenticate(ServerRequestInterface $request, ResponseInterface $response)
     {
         $digest = $this->_getDigest($request);
-        if (empty($digest)) {
+        if ($digest === null) {
             return new Result(null, Result::FAILURE_CREDENTIALS_MISSING);
         }
 
-        $user = $this->_identifier->identify([
-            IdentifierInterface::CREDENTIAL_USERNAME => $digest['username']
-        ]);
+        $identity = $this->_identifier->identify([IdentifierInterface::CREDENTIAL_USERNAME => $digest['username']]);
 
-        if (empty($user)) {
+        if (empty($identity)) {
             return new Result(null, Result::FAILURE_IDENTITY_NOT_FOUND);
         }
 
         if (!$this->validNonce($digest['nonce'])) {
             return new Result(null, Result::FAILURE_CREDENTIALS_INVALID);
         }
 
-        $field = $this->_config['fields'][IdentifierInterface::CREDENTIAL_PASSWORD];
-        $password = $user[$field];
+        $passwordField = $this->getConfig('fields')[IdentifierInterface::CREDENTIAL_PASSWORD];
+        $password = $identity[$passwordField];
 
         $server = $request->getServerParams();
-        if (!isset($server['ORIGINAL_REQUEST_METHOD'])) {
+        if (!array_key_exists('ORIGINAL_REQUEST_METHOD', $server)) {
```

Comment:
isset() is 100x faster and more correct IMO, a key with null has no meaning here and should not do anything

Reply:
Do you have recent data (not from 2014) for PHP 7.2/7.1 regarding this?

Reply:
You can easily come up with this yourself, but its still nonsense to check for keys when the values are the ones we need. And here also nullish values are invalid, all of them.

Reply:
Revert back to `isset` please.